### PR TITLE
[NMS] Subscriber state auto refresh

### PR DESCRIPTION
**Pull request by HannaFar**
## Summary

Add auto refresh for subscriber_state
## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: nms/app/packages/magmalte/app/state/lte/SubscriberState.js [hunk 1]
```diff
@@ -190,12 +193,16 @@ export async function setSubscriberState(props: SubscriberStateProps) {
...
   } = props;
   if (newState) {
-    setSessionState(newState.sessionState);
-    setSubscriberMap(newState.state);
+    setSubscriberMap(newState);
     return;
   }
+  if (newSessionState) {
+    // $FlowIgnore
+    setSessionState(newSessionState.sessionState);
```
Comment (karthiksubraveti) [inline nms/app/packages/magmalte/app/state/lte/SubscriberState.js:204]: Missing return ?

File: nms/app/packages/magmalte/app/state/lte/SubscriberState.js [hunk 2]
```diff
@@ -147,9 +149,12 @@ export default async function InitSubscriberState(
     setSessionState,
     enqueueSnackbar,
   } = props;
-  const subscribers = await FetchSubscribers({networkId, enqueueSnackbar});
-  if (subscribers) {
-    setSubscriberMap(subscribers.subscribers);
+  const susbcriberResponse = await FetchSubscribers({
```
Comment (karthiksubraveti) [inline nms/app/packages/magmalte/app/state/lte/SubscriberState.js]: minor nit: fix typo susbcriberResponse -> subscriberResponse

File: nms/app/packages/magmalte/app/components/ActionTable.js [hunk 3]
```diff
@@ -91,7 +91,7 @@ type ColumnType =
   | 'time'
   | 'currency';
 
-type ActionTableColumn = {
+export type ActionTableColumn = {
```
Comment (karthiksubraveti) [inline nms/app/packages/magmalte/app/components/ActionTable.js]: I don't see this being used. Do we need to export this ?

File: nms/app/packages/magmalte/app/views/subscriber/SubscriberOverview.js [hunk 4]
```diff
@@ -111,7 +71,48 @@ type Props = {
...
+      icon: PeopleIcon,
+    },
+  ];
+
+  return (
+    <>
+      <TopBar
+        header={TITLE}
+        tabs={
+          networkCtx.networkType !== LTE
```
Comment (karthiksubraveti) [inline nms/app/packages/magmalte/app/views/subscriber/SubscriberOverview.js]: I again check this one. Looks like we may have cases where we will use FEG_LTE networks with local subs. We might have to display both the tabs.

File: nms/app/packages/magmalte/app/views/subscriber/SubscriberOverview.js [hunk 5]
```diff
@@ -133,180 +134,37 @@ function JsonDialog(props: Props) {
...
-        )}
-      </div>
-    </>
+export function RenderLink(props: RenderLinkType) {
+  const {relativeUrl, history} = useRouter();
+  const {subscriberConfig, currRow, networkCtx} = props;
+  const imsi = currRow.imsi;
+  return (
+    <div>
+      {subscriberConfig || networkCtx.networkType === FEG_LTE ? (
```
Comment (karthiksubraveti) [inline nms/app/packages/magmalte/app/views/subscriber/SubscriberOverview.js]: Same comment as above.

**Review by karthiksubraveti — COMMENTED** [review]
Thanks for fixing this. Left a few comments.

File: nms/app/packages/magmalte/app/components/ActionTable.js (hunk 3 above)
Comment (HannaFar) [reply nms/app/packages/magmalte/app/components/ActionTable.js]: no I forgot to remove it

File: nms/app/packages/magmalte/app/views/subscriber/SubscriberOverview.js [hunk 6]
```diff
@@ -137,38 +130,25 @@ export function JsonDialog(props: Props) {
...
-          {imsi}
-        </Link>
-      ) : (
-        <>{imsi}</>
-      )}
+      <Link
+        variant="body2"
+        component="button"
+        onClick={() =>
+          // Link to event tab if FEG_LTE network
```
Comment (karthiksubraveti) [inline nms/app/packages/magmalte/app/views/subscriber/SubscriberOverview.js]: remove old comment

File: nms/app/packages/magmalte/app/views/subscriber/SubscriberTable.js [hunk 7]
```diff
@@ -110,6 +118,7 @@ function SubscribersTable(props: WithAlert) {
   return (
     <>
       <div className={classes.dashboardRoot}>
+        <CardTitleRow key="title" icon={SettingsIcon} label={'Subscribers'} />
         <JsonDialog open={jsonDialog} onClose={onClose} imsi={currRow.imsi} />
         <ActionTable
           tableRef={tableRef}
```
Comment (karthiksubraveti) [inline nms/app/packages/magmalte/app/views/subscriber/SubscriberTable.js:124]: In action table, can you add a field `localization` https://github.com/magma/magma/blob/4a92d2280b005d35cc95d7a6896dc863b2fa922c/nms/app/packages/magmalte/app/components/DashboardAlertTable.js#L297 
As shown above. We can set the search string hint through this. We can set it to provide information on what user should search for. For e.g IMSI010101010101

**Review by karthiksubraveti — APPROVED** [review]
Changes LGTM

**Pull request merged**